service/src/main/java/com/commercetools/pspadapter/payone/notification/NotificationProcessorBase.java

```diff
@@ -121,23 +120,20 @@ public final void processTransactionStatusNotification(final Notification notifi
      * the same instance.
      */
     private CompletionStage<Order> updateOrderIfExists(Order order, Payment updatedPayment) {
-        if (order != null) {
-            PaymentState newPaymentState = getPaymentToOrderStateMapper().mapPaymentToOrderState(updatedPayment);
-
-            if (newPaymentState == null && updatedPayment.getPaymentStatus() != null) {
-                logger.warn("Payment [{}] has paymentStatus [{}] which can't be mapped to Order#paymentState. "
-                                + "The order's state remains unchanged.",
-                        updatedPayment.getId(), updatedPayment.getPaymentStatus().getInterfaceCode());
-            }
-
-            // skip update for undefined or unchanged payment state
-            if (newPaymentState != null && !newPaymentState.equals(order.getPaymentState())) {
-                return getOrderService().updateOrderPaymentState(order, newPaymentState);
-            }
-        } else {
-            // This may happen when Payone sends the notification faster then user was redirected back to the shop
-            // and the order has not been created yet.
-            logger.info("Payment [{}] is updated, but respective order is not found!", updatedPayment.getId());
+
+        final PaymentState newPaymentState = getPaymentToOrderStateMapper()
+            .mapPaymentToOrderState(updatedPayment);
```

Comment:
I know it is not much happening here but previously we would not execute this line if order was `null`